ci/setup-build-test-gdk.ps1

```diff
@@ -3,25 +3,38 @@ param(
     [string] $gcs_publish_bucket = "io-internal-infra-unreal-artifacts-production/UnrealEngine",
     [string] $msbuild_exe = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\MSBuild.exe",
     [string] $build_home = (Get-Item "$($PSScriptRoot)").parent.parent.FullName, ## The root of the entire build. Should ultimately resolve to "C:\b\<number>\".
-    [string] $unreal_engine_symlink_dir = "$build_home\UnrealEngine"
+    [string] $unreal_engine_symlink_dir = "$build_home\UnrealEngine",
+    [string] $gym_version_path = "$unreal_engine_symlink_dir\UnrealGDKTestGymVersion.txt"
 )
 
 class TestProjectTarget {
     [ValidateNotNullOrEmpty()][string]$test_repo_url
     [ValidateNotNullOrEmpty()][string]$test_repo_branch
     [ValidateNotNullOrEmpty()][string]$test_repo_relative_uproject_path
     [ValidateNotNullOrEmpty()][string]$test_project_name
+    [ValidateNotNullOrEmpty()][string]$test_gym_version_path
+    [ValidateNotNull()][string]$test_env_override
     
-    TestProjectTarget([string]$test_repo_url, [string]$gdk_branch, [string]$test_repo_relative_uproject_path, [string]$test_project_name) {
+    TestProjectTarget([string]$test_repo_url, [string]$gdk_branch, [string]$test_repo_relative_uproject_path, [string]$test_project_name, [string]$test_gym_version_path, [string]$test_env_override) {
         $this.test_repo_url = $test_repo_url
         $this.test_repo_relative_uproject_path = $test_repo_relative_uproject_path
         $this.test_project_name = $test_project_name
+        $this.test_gym_version_path = $test_gym_version_path
+        $this.test_env_override = $test_env_override
         
-        # If the testing repo has a branch with the same name as the current branch, use that
+        # Resolve the branch to run against. The order of priority is:
+        # envvar > same-name branch as the branch we are currently on > UnrealGDKTestGymVersion.txt > "master".
```

Comment:
I'd say the existence of a version file removes the need to check for a branch of the same name.

Reply:
This is copied from the example project.
The reason for this priority system is so when you are making a change interdependent between two repos, such as TestGyms and GDK (e.g. a new test), then you can have the two PRs "link up" just by using the same branch name, which you would usually do anyway, and you don't have to worry about the version files, which makes it easier to use.

Reply:
I think that is a reasonable argument, but this is in keeping with UnrealGDKExampleProject\ci\generate-pipeline-steps.sh and I prefer to keep things consistent.

Reply:
Let's keep it as is then 👍